### DISCUSSION TITLE
Any plan to support less?

### DISCUSSION THREAD

**Discussion by dungmv** [vercel/next.js#22823] [Ideas]
### Describe the feature you'd like to request

I want to use ant design and ant design mobile with nextjs, but nextjs only has build-in css and sass.

### Describe the solution you'd like

if nextjs has build-in less support, it's very useful

### Describe alternatives you've considered

if nextjs not enough resource to add build-in less support, so you can update less plugin, it isn't updated a long time. it seems outdated

**Comment by haggen** [comment]
Hi @dungmv, I don't know Next.js' plans for LESS integration but I also needed to work with Ant Design so I took matters myself. Here's what I did:

1. Seeing that the LESS plugin was deprecated, that no de facto replacement was available and that there were several gotchas to work with Ant Design I decided to come up with my own solution;
2. Since Next.js has built-in support for SASS which is a preprocessor pretty much like LESS I thought I could start from there, so I dove into Next.js' related code wanting to replicate it for LESS.
3. I've been using this solution for the last few months and it's doing well, taking full advantage of code splitting, tree shaking, CSS modules and every optimization Next.js offers.
4. What I ended up with is to hook into the Webpack config generation on startup, find the rules related to SASS, make a deep copy and then tweak it to apply to LESS. Here's the code; https://gist.github.com/haggen/05dc0d9a9d1e30334ce94fcabf193fe4

A few gotchas:

- I provide an `additionalData` option to LESS parser so I can override antd's styles. It has some hard coded paths specific to my project. You can tweak it and use the feature or drop it altogether, it isn't needed.
- Next.js disables SASS support when you add custom CSS rules; you can potentially disable this safety using the same mechanism I use for LESS at lines 140-155.

**Reply by dungmv** [reply]
after many times to try and research, I gave up. I used tailwind instead of ant design. It's easier to integrate with nextjs

**Reply by JaeWangL** [reply]
Thanks your effort I used your code for my project
But, When I enter my websites

```
./node_modules/antd/lib/style/index.less ((webpack)/css-loader/cjs.js??ref--5-oneOf-12-1!(webpack)/postcss-loader/cjs.js??ref--5-oneOf-12-2!(webpack)/resolve-url-loader??ref--5-oneOf-12-3!./node_modules/less-loader/dist/cjs.js??ref--5-oneOf-12-4!./node_modules/antd/lib/style/index.less)
TypeError: this.getOptions is not a function
```

error is showed

```
  "dependencies": {
    "@ant-design/icons": "^4.5.0",
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.1.5",
    "@tanem/react-nprogress": "^3.0.58",
    "antd": "^4.14.0",
    "axios": "^0.21.1",
    "babel-plugin-import": "^1.13.3",
    "bootstrap": "^4.6.0",
    "immer": "^8.0.1",
    "less": "^4.1.1",
    "next": "^10.0.8",
    "next-i18next": "^8.1.1",
    "next-redux-wrapper": "^6.0.2",
    "next-seo": "^4.20.0",
    "react": "^17.0.1",
    "react-bootstrap": "^1.5.2",
    "react-dom": "^17.0.1",
    "react-fast-compare": "^3.2.0",
    "react-icons": "^4.2.0",
    "react-redux": "^7.2.2",
    "redux": "^4.0.5",
    "redux-saga": "^1.1.3",
    "sass": "^1.32.8"
  },
```

Do you know what this problem is??

**Reply by dungmv** [reply]
@JaeWangL  maybe javascript not enable when compiling less
